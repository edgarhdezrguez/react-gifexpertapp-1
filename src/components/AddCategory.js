import React, {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {
    // const [categories, setCategories] = catgs;
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        // console.log('handleInputChange call');
    };

    const handleSubmit = (e) =>{
        e.preventDefault();
        
        // console.log('handleSubmit called', inputValue);

        if (inputValue.trim().length > 0){
            setCategories(categories =>  [inputValue, ...categories]);
            setInputValue('');
            // console.log('in if');
        } else {
            //alert("Value dont be empty");
            console.log('in else');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                value={inputValue}
                onChange={handleInputChange}
                name="addSerie" 
                id="serie2Add"/>
            {/* <button onClick={handleAdd}> Add </button> */}
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
