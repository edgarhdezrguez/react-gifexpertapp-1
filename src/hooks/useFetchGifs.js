import { useEffect, useState } from "react"
import { getGifts } from "../helpers/getGifts";


export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifts(category)
        .then( imgs => setState({data: imgs, loading:false}))
        .catch(console.log );
    }, [category]);

    return state; // { data: [], loading: true}

};