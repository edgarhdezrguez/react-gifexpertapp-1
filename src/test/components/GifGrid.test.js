import React from 'react';
import { shallow, configure } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { useFetchGifs } from '../../hooks/useFetchGifs';
configure({ adapter: new Adapter() });
jest.mock('../../hooks/useFetchGifs');

describe('Tests in GifGrid Component.', () => {

    const value = 'Goku';
    

    test('should be show the component correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={value} />);
        expect(wrapper).toMatchSnapshot();

    })
    
    test('should be show items when images load with useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:8080/any/thing.jpg',
            title: 'Anything'
        },
        {
            id: 'ABCD',
            url: 'https://localhost:8080/any/thing.jpg',
            title: 'Anything 2'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={value} />);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    



})
