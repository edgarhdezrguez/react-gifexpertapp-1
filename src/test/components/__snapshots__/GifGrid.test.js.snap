// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid Component. should be show the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
    key="Goku"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
